f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py

```diff
@@ -1755,3 +1807,78 @@ def delete(self, l7rule, service, **kwargs):
         else:
             # Update LTM policy
             self._update_ltm_policy(self._l7policy, service)
+
+
+class ACLGroupManager(ResourceManager):
+
+    def __init__(self, driver, **kwargs):
+        super(ACLGroupManager, self).__init__(driver)
+        self._resource = "ACLGroup"
+        self.resource_helper = resource_helper.BigIPResourceHelper(
+            resource_helper.ResourceType.internal_data_group)
+
+    def _create_payload(self, resource):
+        """Convert neutron lbaas acl_group to bigip data_group"""
+
+        partition = "Common"
+        name = "acl_" + resource["id"]
+        data_group_type = "ip"
+        rules = resource.get("acl_rules_detail")
+        data_group_rules = {}
+
+        if rules:
+            # do we need to put data as rule['id'] here?
+            data_group_rules = [
+                {"name": rule["ip_address"], "data": ""}
+                for rule in rules]
+
+        payload = {
+            "name": name,
+            "partition": partition,
+            "type": data_group_type,
+            "records": data_group_rules
+        }
+
+        return payload
+
+    @log_helpers.log_method_call
+    def create(self, resource, **kwargs):
```

Comment:
Not quite sure why it does not reuse the implementation in ResourceManager. Is there any specific purpose?